corehq/apps/api/resources/messaging_event/view.py

```diff
@@ -26,7 +28,14 @@ def messaging_events(request, domain, event_id=None):
     """Despite it's name this API is backed by the MessagingSubEvent model
     which has a more direct relationship with who the messages are being sent to.
     Each event may have more than one actual message associated with it.
+
+    Due to performance issues, this API only allows one usage at a time
     """
+
+    # if the request does not acquire the lock in one minute, return a rate limiting response
+    messaging_events_lock = get_redis_lock('MESSAGING_EVENTS_API_LOCK', timeout=60*10, name='api')
+    if not messaging_events_lock.acquire(blocking=True, timeout=60):
```

Comment:
I dug for a couple minutes, and I believe the parameter is called `blocking_timeout`. (Incidentally it seems like you can also specify it in the `get_redis_lock` call to set a default value, but I do like it better in the `acquire` call.)

Reply:
Huh, django redis, uses the same interface as `threading.Lock`, which uses `timeout`, but it looks like this lock is directly from the redis python library. That is my bad.